kv: TestRangeLookupOptionOnReverseScan stuck on gossip network on drain

```
I160509 22:03:04.911557 stopper.go:352  draining; tasks left:
5      gossip/infostore.go:286
1      gossip/gossip.go:750
I160509 22:03:04.912041 gossip/gossip.go:918  starting client to 127.0.0.1:46570
panic: test timed out after 1m10s
```

Interestingly, while the CI script posted the data to create an issue and there's no obvious error, no issue was created.

https://circleci.com/gh/cockroachdb/cockroach/17533

--- a/rpc/context.go
+++ b/rpc/context.go
@@ -188,6 +188,14 @@ func (ctx *Context) runHeartbeat(cc *grpc.ClientConn, remoteAddr string) error {
 	var heartbeatTimer timeutil.Timer
 	defer heartbeatTimer.Stop()
 	for {
+		// If we should stop, return immediately. Note that we check this
+		// at the beginning and end of the loop because we may 'continue'
+		// before reaching the end.
+		select {
+		case <-ctx.Stopper.ShouldStop():
+			return nil
+		default:
+		}
 		sendTime := ctx.localClock.PhysicalTime()
 		response, err := ctx.heartbeat(heartbeatClient, request)
 		if err != nil {

